docs/content/querying/dimensionspecs.md

```diff
@@ -252,3 +252,16 @@ A null dimension value can be mapped to a specific value by specifying the empty
 This allows distinguishing between a null dimension and a lookup resulting in a null.
 For example, specifying `{"":"bar","bat":"baz"}` with dimension values `[null, "foo", "bat"]` and replacing missing values with `"oof"` will yield results of `["bar", "oof", "baz"]`.
 Omitting the empty string key will cause the missing value to take over. For example, specifying `{"bat":"baz"}` with dimension values `[null, "foo", "bat"]` and replacing missing values with `"oof"` will yield results of `["oof", "oof", "baz"]`.
+
+### Filtering DimensionSpecs
+These are only valid for multi-valued dimensions. They take a delegate DimensionSpec and a filtering criteria, multiple values list of dimension is filtered as per given criteria.
```

Comment:
The difference between the "predicate" style of filter and the whitelist/blacklist is one of optimization.  With a predicate, you have to scan through the entire dictionary in order to figure out what matches and what doesn't.  With a whitelist/blacklist, you can lookup only the things in the list.